### PR TITLE
docs: support argument overrides

### PR DESCRIPTION
**Pull request by yury-s**
* Supported parameter overrides - default definition can can replaced by a language specific one. Override is matched with the default definition by either name or alias. This ensures that overriding parameter always goes in the same order as the default one,.
* Inverted `pageFunction` parameter definition: js is now an override, python version became default
* Use separate if  for `force_expression` in python to avoid confusion

Corresponding change to the docs site: https://github.com/microsoft/playwright.dev/pull/93

### CODE REVIEW

File: docs/src/api/javascript.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+### param: ElementHandle.$eval.pageFunction = %%-js-evalonselector-pagefunction-%%
```
Comment (pavelfeldman) [inline docs/src/api/javascript.md:1]: We should not allow overriding by alias, this is confusing. Let's always use original name instead.

File: docs/src/api/params.md [hunk 2]
```diff
@@ -154,6 +154,38 @@ Toggles bypassing page's Content-Security-Policy.
 
 Sets a consistent viewport for each page. Defaults to an 1280x720 viewport. `null` disables the default viewport.
 
+## evaluate-expression
+* langs:
+  - alias-js: pageFunction
```
Comment (pavelfeldman) [inline docs/src/api/params.md:159]: Follow up: we won't need alias here since we have enough info in the override itself.

File: utils/doclint/api_parser.js [hunk 3]
```diff
@@ -122,7 +122,18 @@ class ApiParser {
     if (!method)
       throw new Error('Invalid method ' + match[2] + '.' + match[3]);
     if (match[1] === 'param') {
-      method.argsArray.push(this.parseProperty(spec));
+      const arg = this.parseProperty(spec);
+      const existingArg = method.argsArray.find(m => m.name === arg.name || m.langs.aliases && Object.values(m.langs.aliases).indexOf(arg.name) !== -1);
```
Comment (pavelfeldman) [inline utils/doclint/api_parser.js:126]: drop the aliases match here.

File: utils/doclint/api_parser.js [hunk 4]
```diff
@@ -122,7 +122,18 @@ class ApiParser {
...
+      if (existingArg) {
+        for (const lang of arg.langs.only) {
+          if (existingArg.name !== arg.name && existingArg.langs.aliases[lang] !== arg.name)
```
Comment (pavelfeldman) [inline utils/doclint/api_parser.js:129]: Remove this check, it is to override in anextensible manner without the core knowing.

File: utils/doclint/api_parser.js [hunk 5]
```diff
@@ -302,7 +313,8 @@ function extractLangs(spec) {
     return {
       only: only ? only.split(',') : undefined,
       aliases,
-      types: {}
+      types: {},
```
Comment (pavelfeldman) [inline utils/doclint/api_parser.js:316]: Do you want to migrate existingMemeber to `overrides` as well?

File: utils/doclint/generateApiJson.js [hunk 6]
```diff
@@ -35,7 +35,7 @@ const PROJECT_DIR = path.join(__dirname, '..', '..');
   });
   documentation.generateSourceCodeComments();
   const result = serialize(documentation);
-  fs.writeFileSync(path.join(PROJECT_DIR, 'api.json'), JSON.stringify(result));
+  fs.writeFileSync(path.join(PROJECT_DIR, 'api.json'), JSON.stringify(result, null, 2));
```
Comment (pavelfeldman) [inline utils/doclint/generateApiJson.js:38]: revert this

File: utils/doclint/documentation.js [hunk 7]
```diff
@@ -322,11 +323,12 @@ Documentation.Member = class {
         continue;
       if (arg.langs.aliases && arg.langs.aliases[lang])
         arg.alias = arg.langs.aliases[lang];
-      arg.filterForLanguage(lang);
-      arg.type.filterForLanguage(lang);
-      if (arg.name === 'options' && !arg.type.properties.length)
+      const overriddenArg = (arg.langs.overrides && arg.langs.overrides[lang]) || arg;
+      overriddenArg.filterForLanguage(lang);
+      overriddenArg.type.filterForLanguage(lang);
```
Comment (pavelfeldman) [inline utils/doclint/documentation.js:328]: Drive-by: looks like we are already doing this in line 319, so drop this line.

**Review by pavelfeldman — APPROVED** [review]

File: docs/src/api/javascript.md (hunk 1 above)
Comment (yury-s) [reply docs/src/api/javascript.md]: Done.

File: docs/src/api/params.md (hunk 2 above)
Comment (yury-s) [reply docs/src/api/params.md]: Done.

File: utils/doclint/api_parser.js (hunk 4 above)
Comment (yury-s) [reply utils/doclint/api_parser.js]: Done.

File: utils/doclint/api_parser.js (hunk 3 above)
Comment (yury-s) [reply utils/doclint/api_parser.js]: Done.

File: utils/doclint/documentation.js (hunk 7 above)
Comment (yury-s) [reply utils/doclint/documentation.js]: Done.

File: utils/doclint/generateApiJson.js (hunk 6 above)
Comment (yury-s) [reply utils/doclint/generateApiJson.js]: Done.
